### PR TITLE
[LLVM][DWARF] Make dwarf::getDebugNamesBucketCount return a pair.

### PR DESCRIPTION
**Pull request by cmtice**
llvm::dwarf::getDebugNamesBucketCount directly returns the bucket count, via return statement, but it also returns the hash count via a parameter. This changes the function to return them both as a std::pair, in the return statement. It also changes the name of the function to make it clear it returns both values.

### CODE REVIEW

File: llvm/include/llvm/BinaryFormat/Dwarf.h [hunk 1]
```diff
@@ -616,20 +616,21 @@ enum AcceleratorTable {
 // Uniquify the string hashes and calculate the bucket count for the
 // DWARF v5 Accelerator Table. NOTE: This function effectively consumes the
 // 'hashes' input parameter.
-inline uint32_t getDebugNamesBucketCount(MutableArrayRef<uint32_t> hashes,
-                                         uint32_t &uniqueHashCount) {
-  uint32_t BucketCount = 0;
+inline std::pair<uint32_t, uint32_t> getDebugNamesBucketAndHashCount(
+    MutableArrayRef<uint32_t> hashes) {
+  uint32_t uniqueHashCount = 0;
```
[the remark this answers is not in the document]
Comment (cmtice) [reply llvm/include/llvm/BinaryFormat/Dwarf.h]: Done. :-)

**Review by felipepiovezan — APPROVED** [review]
LGTM!

File: llvm/include/llvm/BinaryFormat/Dwarf.h [hunk 2]
```diff
@@ -616,20 +616,20 @@ enum AcceleratorTable {
 // Uniquify the string hashes and calculate the bucket count for the
 // DWARF v5 Accelerator Table. NOTE: This function effectively consumes the
 // 'hashes' input parameter.
-inline uint32_t getDebugNamesBucketCount(MutableArrayRef<uint32_t> hashes,
-                                         uint32_t &uniqueHashCount) {
-  uint32_t BucketCount = 0;
+inline std::pair<uint32_t, uint32_t>
+getDebugNamesBucketAndHashCount(MutableArrayRef<uint32_t> hashes) {
+  uint32_t bucketCount = 0;
```
Comment (JDevlieghere) [inline llvm/include/llvm/BinaryFormat/Dwarf.h:621]: Why did you change the case? Doesn't LLVM use UperCamelCase for variables?

Comment (felipepiovezan) [reply llvm/include/llvm/BinaryFormat/Dwarf.h:621]: good point, I had missed this

Comment (cmtice) [reply llvm/include/llvm/BinaryFormat/Dwarf.h:621]: I thought LLVM variables were supposed to start with lower case?  So I thought I was fixing a problem? I can switch it back if you think I should...

**Comment by MaskRay** [comment]
I suggested `pair` in the previous patch. Thanks for changing `getDebugNamesBucketAndHashCount`

File: llvm/include/llvm/BinaryFormat/Dwarf.h (hunk 2 above)
Comment (JDevlieghere) [reply llvm/include/llvm/BinaryFormat/Dwarf.h:621]: You're probably thinking of functions, which (unlink LLDB) use `lowerCamelCase`: https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly

Comment (cmtice) [reply llvm/include/llvm/BinaryFormat/Dwarf.h]: Thanks! Done.

**Review by JDevlieghere — APPROVED** [review]

File: llvm/lib/CodeGen/AsmPrinter/AccelTable.cpp [hunk 3]
```diff
@@ -38,7 +38,9 @@ void AccelTableBase::computeBucketCount() {
   for (const auto &E : Entries)
     Uniques.push_back(E.second.HashValue);
 
-  BucketCount = llvm::dwarf::getDebugNamesBucketCount(Uniques, UniqueHashCount);
+  auto Counts = llvm::dwarf::getDebugNamesBucketAndHashCount(Uniques);
```
Comment (MaskRay) [inline llvm/lib/CodeGen/AsmPrinter/AccelTable.cpp:41]: This can be simplified to `std::tie(BucketCount, UniqueHashCount)`

**Review by MaskRay — APPROVED** [review]

**Comment by cmtice** [comment]
The buildkit build failure appears to have not relation to this change. Does anybody know how I can force the build test to re-run, so that I can get a green "Squash and merge" button?

**Comment by felipepiovezan** [comment]
> The buildkite build failure appears to have no relation to this change. Does anybody know how I can force the build test to re-run, so that I can get a green "Squash and merge" button?

You can still click the button, the failure seems unrelated indeed

File: llvm/lib/CodeGen/AsmPrinter/AccelTable.cpp (hunk 3 above)
Comment (dwblaikie) [reply llvm/lib/CodeGen/AsmPrinter/AccelTable.cpp:41]: +1 to this - @cmtice perhaps you could make this change as a follow-up commit. No need to send it for review. (but please follow up on this comment thread and mention the commit hash of the change once it's upstream)

Comment (MaskRay) [reply llvm/lib/CodeGen/AsmPrinter/AccelTable.cpp:41]: I made the simplification already. Sometimes reviewers may click "Accept" and expect some minor code suggestions will be adopted before landing. It's worth paying attention to these comments.

(For both #82153 and this #83047, my comments were created before the patch landed. Don't worry. We all made mistakes.)

Comment (dwblaikie) [reply llvm/lib/CodeGen/AsmPrinter/AccelTable.cpp:41]: ah, https://github.com/llvm/llvm-project/commit/7b0b64a7016d67d6d8c538fe406a2ffd93c11602 cool, thanks!
